torch/_dynamo/eval_frame.py

```diff
@@ -91,55 +91,116 @@ class Unset(Enum):
 
 class OptimizedModule(torch.nn.Module):
     """
-    Wraps the original nn.Module object and later patches its
-    forward method to optimized self.forward method.
+    Wraps the original nn.Module object and later patches its `forward` method to optimized `self.forward` method.
+    When calling `torch.compile(mod)` the output is an `OptimizedModule`
+    Compilation occurs the first time `forward()` is called
+
+    Note:
+    - The OptimizedModule and original module share `state_dict` so you can run `torch.save(opt_model.state_dict())`
+
     """
 
     def __init__(self, mod: torch.nn.Module, dynamo_ctx):
+        """
+        Initializes an instance of OptimizedModule.
```

Comment:
This is the return value of torch.compile() and not intended to be constructed by end users.  I think most of this shouldn't end up in the docs.

Reply:
Yeah I guess now that pickling the optimized module will work. I don't see any reason people need to know about it

Reply:
`__setattr__` just behaves exactly the same as `nn.Module.__setattr__`

Reply:
What I worry more about is whether people need to understand what an optimized module is. If yes then probably should document it but tell people they should never create it directly

Reply:
Retying answer, OK makes sense I think if the intent is that users need to neither know about `OptimizedModule` because they can `__setattr__`  and because they can pickle it because of my other PR then it shouldn't be public